apps/total-typescript/src/lib/bookmarks.ts

```diff
@@ -0,0 +1,164 @@
+import {z} from 'zod'
+import {prisma} from '@skillrecordings/database'
+
+export const BookmarkInputSchema = z.object({
+  module: z.string(),
+  section: z.object({
+    title: z.string(),
+    slug: z.string(),
+  }),
+  resource: z.object({
+    id: z.string(),
+    children: z.string(),
+  }),
+  userId: z.string(),
+})
+
+export const ReturnedBookmarkSchema = z.object({
+  id: z.string(),
+  module: z.string(),
+  section: z.object({
+    title: z.string(),
+    slug: z.string(),
+  }),
+  resource: z.object({
+    id: z.string(),
+    children: z.string(),
+  }),
+})
+
+export type Bookmark = z.infer<typeof ReturnedBookmarkSchema>
+export type BookmarkInput = z.infer<typeof BookmarkInputSchema>
+
+export async function getBookmarksForUser(userId: string) {
+  try {
+    const bookmarks = await prisma.bookmark.findMany({
+      select: {
+        id: true,
+        module: true,
+        section: true,
+        resource: true,
+      },
+      where: {
+        userId,
+      },
+    })
+    return bookmarks
+  } catch (error) {
+    console.error('Error getting bookmarks for user', error)
+    throw new Error('Error getting bookmarks for user')
+  }
+}
+
+export async function getBookmarkByResourceId({
+  userId,
+  resourceId,
+}: {
+  userId: string
+  resourceId: string
+}) {
+  try {
+    const bookmark = await prisma.bookmark.findFirst({
+      select: {
+        id: true,
+        module: true,
+        section: true,
+        resource: true,
+      },
+      where: {
+        userId,
+        resource: {
+          path: '$.id',
+          equals: resourceId,
+        },
```

Comment:
filtering on JSON values is a little weird but this is what prisma docs recommend https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types/working-with-json-fields#filter-on-a-json-field-advanced